src/wcs1_0/src/main/java/org/geoserver/wcs/DefaultWebCoverageService100.java

```diff
@@ -425,10 +433,34 @@ public GridCoverage[] getCoverage(final GetCoverageType request) {
             //
             coverage = reader.read(readParameters);
             if ((coverage == null) || !(coverage instanceof GridCoverage2D)) {
-                throw new IOException(
-                        "No raster data found in the request (it may be that "
-                                + "the request bbox is outside of the coverage area, or that the filters used "
-                                + "match no portions of it.");
+                Integer[] values = new Integer[1];
```

Comment:
Better to check the datatype of the CoverageInfo and use proper number type based on that.
See the meta.dimensions elements.